packages/piro/test/Main_AnalysisDriver_Tpetra.cpp

```diff
@@ -153,21 +167,22 @@ int main(int argc, char *argv[]) {
       RCP<Thyra::VectorBase<double>> p;
       status = Piro::PerformAnalysis(*piro, analysisParams, p);
 
-      Thyra::DetachedVectorView<double> p_view(p);
-      double p_exact[2] = {1,3};
-      double tol = 1e-6;
-
-      double l2_diff = std::sqrt(std::pow(p_view(0)-p_exact[0],2) + std::pow(p_view(1)-p_exact[1],2));
-      if(l2_diff > tol) {
-        status+=100;
-        if (Proc==0) {
-          std::cout << "\nPiro_AnalysisDrvier:  Optimum parameter values are: {"
-              <<  p_exact[0] << ", " << p_exact[1] << "}, but computed values are: {"
-              <<  p_view(0) << ", " << p_view(1) << "}." <<
-              "\n                      Difference in l2 norm: " << l2_diff << " > tol: " << tol <<   std::endl;
+      if (Teuchos::nonnull(p)) { //p might be null if the packages ROL, Moocho, OptiPack are not enabled
```

Comment:
It sounds like you might want the test to fail if any of those packages is not enabled.  Is that true?

Reply:
I copied this test from Epetra-based test Main_AnalysisDriver, and the logic there is that if some packages are not enabled the test will not run the parts requiring those tests without failing. I decided to keep that logic, but we can change it.

Reply:
That's fine -- I just wanted to check.  Thanks!